src/rpc/ws_client/mod.rs

```diff
@@ -108,271 +105,17 @@ impl HandleMessage for SubscriptionHandler {
 
 		match value["id"].as_str() {
 			Some(_idstr) => {},
-			_ => {
-				// subscriptions
-				debug!("no id field found in response. must be subscription");
-				debug!("method: {:?}", value["method"].as_str());
-				match value["method"].as_str() {
-					Some("state_storage") => {
-						let changes = &value["params"]["result"]["changes"];
-						match changes[0][1].as_str() {
-							Some(change_set) => {
-								if let Err(SendError(e)) = result.send(change_set.to_owned()) {
-									debug!("SendError: {:?}. will close ws", e);
-									out.close(CloseCode::Normal)?;
-								}
-							},
-							None => println!("No events happened"),
-						};
-					},
-					Some("chain_finalizedHead") => {
-						let head = serde_json::to_string(&value["params"]["result"])
-							.map_err(|e| Box::new(RpcClientError::Serde(e)))?;
-
-						if let Err(e) = result.send(head) {
-							debug!("SendError: {}. will close ws", e);
-							out.close(CloseCode::Normal)?;
-						}
-					},
-					_ => error!("unsupported method"),
+			None => {
+				let answer = serde_json::to_string(&value["params"]["result"])
+					.map_err(|e| Box::new(RpcClientError::Serde(e)))?;
+
+				if let Err(e) = result.send(answer) {
+					// This may happen if the receiver has unsubscribed.
+					trace!("SendError: {}. will close ws", e);
+					out.close(CloseCode::Normal)?;
 				}
 			},
 		};
 		Ok(())
 	}
 }
-
-#[derive(Default, Debug, PartialEq, Eq, Clone)]
-pub struct SubmitOnlyHandler;
-
-impl HandleMessage for SubmitOnlyHandler {
-	type ThreadMessage = RpcMessage;
-
-	fn handle_message(
-		&self,
-		msg: Message,
-		out: Sender,
-		result: ThreadOut<Self::ThreadMessage>,
-	) -> WsResult<()> {
-		let retstr = msg.as_text()?;
-		debug!("got msg {}", retstr);
-		match result_from_json_response(retstr) {
-			Ok(val) => end_process(out, result, Ok(Some(val))),
-			Err(e) => end_process(out, result, Err(e)),
-		}
-	}
-}
-
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct SubmitAndWatchHandler {
```

Comment:
Obsolete. SubmitAndWatch is now handlded as a simple Subscription that returns a `TransactionStatus`.